### PR TITLE
feat(dedicated.vmware) : custom nsx edges sizing

### PR DESCRIPTION
**Pull request by dectotam**
| Question         | Answer
| ---------------- | ---
| Branch?          | master
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Feat MANAGER-12763
| License          | BSD 3-Clause

- [ ] ~~Try to keep pull requests small so they can be easily reviewed.~~
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

Full feature : custom nsx edges sizing

### Translations
- [x] CMT-9212

### CODE REVIEW

**Review by qpavy — DISMISSED** [review]

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/dashboard/manage-nsx-edges/dedicatedCloud-datacenter-manage-nsx.controller.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    this.DedicatedCloud = DedicatedCloud;
+    this.$translate = $translate;
+    this.coreURLBuilder = coreURLBuilder;
+    this.NSX_RESOURCES = NSX_RESOURCES;
+    this.TRACKING_SUFFIX = TRACKING_SUFFIX;
+  }
+
+  $onInit() {
+    this.loading = false;
+    this.nsx_sizes = Object.keys(EDGES_SIZES);
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/dashboard/manage-nsx-edges/dedicatedCloud-datacenter-manage-nsx.controller.js:19]:
```suggestion
    this.nsxSizes = Object.keys(EDGES_SIZES);
```

**Review by anooparveti — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/dashboard/manage-nsx-edges/dedicatedCloud-datacenter-manage-nsx.controller.js (hunk 1 above)
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/dashboard/manage-nsx-edges/dedicatedCloud-datacenter-manage-nsx.controller.js:19]:
```suggestion
    this.nsxSizes = Object.keys(EDGES_SIZES);
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/dashboard/dedicatedCloud.datacenter.dashboard.routes.js [hunk 2]
```diff
@@ -8,6 +8,72 @@ export default /* @ngInject */ ($stateProvider) => {
...
+          return DedicatedCloud.getDatacenterNsxtEdgesScalingCapabilities(
+            serviceName,
+            datacenterId,
+          )
+            .then(({ data }) => data?.size)
+            .catch(() => []);
+        },
+        goBackToDashboard: /* @ngInject */ (
+          $state,
+          $timeout,
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/dashboard/dedicatedCloud.datacenter.dashboard.routes.js:57]: **nitpick:** Remove the unused `$timeout`
```suggestion
```

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/dashboard/manage-nsx-edges/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      {
+        lazyLoad: ($transition$) => {
+          const $ocLazyLoad = $transition$.injector().get('$ocLazyLoad');
+          return import('./dedicatedCloud-datacenter-manage-nsx.module').then(
+            (mod) => {
+              $ocLazyLoad.inject(mod.default || mod);
+            },
+          );
+        },
+      },
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/dashboard/manage-nsx-edges/index.js:21]: **nitpick:** Add URL.
```suggestion
      {
        url: '/nsx',
        lazyLoad: ($transition$) => {
          const $ocLazyLoad = $transition$.injector().get('$ocLazyLoad');
          return import('./dedicatedCloud-datacenter-manage-nsx.module').then(
            (mod) => {
              $ocLazyLoad.inject(mod.default || mod);
            },
          );
        },
      },
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/dashboard/manage-nsx-edges/dedicatedCloud-datacenter-manage-nsx.html [hunk 4]
```diff
@@ -0,0 +1,104 @@
+<oui-header
+    heading="{{'dedicatedCloud_manage_nsx_edge_title' | translate}}"
+    description="{{ 'dedicatedCloud_tab_datacenter_global_information' | translate}}"
+>
+    <oui-back-button data-on-click="$ctrl.goBack()"> </oui-back-button>
```
Comment (JacquesLarique) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/dashboard/manage-nsx-edges/dedicatedCloud-datacenter-manage-nsx.html:5]: No cancel method (to handle tracking) ?

Comment (dectotam) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/dashboard/manage-nsx-edges/dedicatedCloud-datacenter-manage-nsx.html:5]: A call to trackClick is made inside the goback function in resolve of route file :
```
goBack: /* @ngInject */ (trackClick, goBackToDashboard) => () => {
    trackClick('::cancel');
    return goBackToDashboard();
}
```

**Review by JacquesLarique — APPROVED** [review]

**Review by JacquesLarique — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**